### PR TITLE
Document modem subsystem

### LINKED ISSUE #62327: Add documentation for new modem subsystem
**Is your enhancement proposal related to a problem? Please describe.**

The recently added modem subsystem does not show up in the documentation, making it potentially difficult to know of its existence in the first place, and to learn how to use it for folks who'd be looking for that information.

**Describe the solution you'd like**

* Add documentation in OS Services section of the documentation
* The Doxygen documentation for chat, cmux, etc API, needs to be completed, in particular add groups and make sure they are "mounted" properly in a relevant parent group (`connectivity`?)
   * Please make sure that internal stuff is flagged appropriately (`@cond INTERNAL_HIDDEN`) so that it does not show up in the generated docs, and that all other public API has documentation.
* Update the cellular_modem sample using the change introduced in #62029 (when/if merged) so that it references the doxygen group(s) that are relevant.

**Describe alternatives you've considered**

N/A

**Additional context**

Sligthly related: functions that have multiple return values should use `@retval`, not `@returns` in their Doxygen docstring.

Thanks!

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
Adds documentation for the new modem subsystem along with an entry in the release notes for 3.5

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/services/modem/index.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.. _modem:
+
+Modem modules
+#############
+
+This service provides modules necessary to communicate with modems.
+
+Modems are self-contained devices which implement the hardware and
```
Comment (kartben) [inline doc/services/modem/index.rst:8]: nit:
```suggestion
Modems are self-contained devices that implement the hardware and
```

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks Bjarki!
Spotted a small typo

File: doc/services/modem/index.rst (hunk 1 above)
Comment (bjarki-trackunit) [reply doc/services/modem/index.rst]: corrected

**Review by kartben — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by bjarki-trackunit** [comment]
I just rebased it :D

**Comment by kartben** [comment]
> I just rebased it :D

Darn, I should have added my newly found nits then :D

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by bjarki-trackunit** [comment]
@kartben I will happily fix them after this is merged so I don't have to race against the release notes :)

**Review by jhedberg — APPROVED** [review]

**Pull request merged**